### PR TITLE
Reader: For testing, add support for feed search and followed search on wp.com/following/edit

### PR DESCRIPTION
**Pull request by felixcheung**
This is the placeholder test UI to call the new WPCOM API for feed search and followed search.

There are two entrypoints. The Top Box (aka feed search) provides a way to search for external RSS feeds. Previously it only tries to parse the URL you enter, now it does Instant Search.

The second opens when you click on the search magnifying glass button – the Followed Box provides a way to search for blogs/sites/feeds you have followed. The result is sorted by the es query score.

New behavior is under feature tag 'reader/follow-feed-search'
To run: $ ENABLE_FEATURES=reader/follow-feed-search make run

Known issues:
~~1. I couldn’t figure out how to get multiple results for the Top Box (aka feed search) to show up; only one is there right now (the last one I think, because of how the ListItem is rendered) – instead you can see the full result list in the Javascript console.~~ (fixed, see below)
2. UI is very rough – again the goal is to make it accessible. It does not represent any real user experience.
3. Code changes is taking a bunch of shortcuts to get it to work. FeedSubscriptionStore and associates are fairly tightly integrated so instead here we have a separate path added to access the WPCOM API directly.

cc: @gibrown @bluefuton

Test live: https://calypso.live/?branch=try/reader/follow-feed-search

### CODE REVIEW

**Comment by bluefuton** [comment]
Nicely done @felixcheung!

> 1. I couldn’t figure out how to get multiple results for the Top Box (aka feed search) to show up; only one is there right now (the last one I think, because of how the ListItem is rendered) – instead you can see the full result list in the Javascript console.

Each ListItem needs a unique key to be rendered successfully. At the moment you're running into this error:

screen shot 2016-09-01 at 13 11 27

**Comment by felixcheung** [comment]
ah thanks very much for the tip, @bluefuton.
I had the key but it seemed the field I'm using was not unique enough (some had "undefined").

Now it looks like all items are added (they are visible briefly) but they are stacked on top of each other, at the exact same position?

File: client/reader/following-edit/subscribe-form.jsx [hunk 1]
```diff
@@ -129,11 +160,23 @@ var FollowingEditSubscribeForm = React.createClass( {
...
+			if ( ! config.isEnabled( 'reader/follow-feed-search' ) ) {
+				searchResult = ( <FollowingEditSubscribeFormResult
+					isValid={ isWellFormedFeedUrl }
+					url={ searchString }
+					onFollowToggle={ handleFollowToggle } />
+				);
+			} else if ( this.state.newFeeds ) {
+				console.log( this.state.newFeeds );
+				searchResult = this.state.newFeeds.map( function( item ) {
+				 		return ( <div className="subscription-list-item__header-content"
```
Comment (gibrown) [inline client/reader/following-edit/subscribe-form.jsx]: I suspect the issue for showing multiple results is here combined with the absolute positioning at line 145 of style.scss in the same dir. By removing this extra div and disabling the position:absolute in my browser I go multiple results to show up (ugly but maybe workable and gets you on the right path?).

**Comment by gibrown** [comment]
I think when I tried this the other day I got confused by there only being one result. I think if we can get the multiple results working and looking good enough then we can probably get this onto wpcalypso for others to try out and so we can work on the search query a bit more.

**Comment by felixcheung** [comment]
Thanks for the tip! Pushed a fix - bascially I'm setting className to `is-search-result` which is fix position.

Now it looks like this (not great, but it's the full result)
image

**Comment by lancewillett** [comment]
Hi @felixcheung Checking on this one — it is stalled or pending more code fixes and review?

**Comment by felixcheung** [comment]
Hi - I guess it's now up to @bluefuton or @gibrown whether they want to take this in 😄

**Comment by gibrown** [comment]
Ya, we were just planning out having someone pick it up in the next few weeks to see if we can ship it with the rest of the refresh.

**Comment by lancewillett** [comment]
This one needs a conflict resolution and decision to merge or not.

**Comment by gibrown** [comment]
@samouri is now actively working on this.

**Comment by samouri** [comment]
We can close this PR though. I'm working on it in #10284

**Pull request closed without merging**